src/Site/Controller/Redirect.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Site\Controller;
+
+use Api\Library\Shared\Palaso\Exception\UserUnauthorizedException;
+use Api\Library\Shared\SilexSessionHelper;
+use Api\Model\Shared\ProjectModel;
+use Api\Model\Shared\UserModel;
+use Silex\Application;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class Redirect extends App
+{
+    /**
+     * @param Request $request
+     * @param Application $app
+     * @param $appName
+     * @param string $projectId
+     * @return Response
+     */
+    public function view(
+        /** @noinspection PhpUnusedParameterInspection */
+        Request $request, Application $app, $appName, $projectId = ''
+    ) {
+        $model = new AppModel($app, $appName, $this->website, $projectId);
+        try {
+            $this->setupBaseVariables($app);
+        } catch (\Exception $e) {
+            return $app->redirect('/auth/logout');
+        }
+        try {
+            if ($model->requireProject) {
+                if ($model->isPublicApp) {
+                    $model->projectId = SilexSessionHelper::requireValidProjectIdForThisWebsite($model->app, $this->website, $model->projectId);
+                } else {
+                    $model->projectId =
+                        SilexSessionHelper::requireValidProjectIdForThisWebsiteAndValidateUserMembership($model->app, $this->website, $model->projectId);
+                }
+            }
+        } catch (UserUnauthorizedException $e) {
+            if (SilexSessionHelper::getUserId($app)) {
+                // User tried to access project they're not a member of, so show them projects view so they can pick a different one
+                return $app->redirect("/app/projects");
+            }
+            return $app->redirect('/auth/logout');
+        } catch (\Exception $e) {
+            return $app->redirect('/auth/logout');
+        }
+        if ($model->projectId) {
+            try {
+                $project = new ProjectModel($model->projectId);
+                if ($project && $project->appName) {
+                    return $app->redirect("/app/$project->appName/$model->projectId");
+                }
+            } catch (\Exception $e) {
+                return $app->redirect('/auth/logout');
+            }
+        }
```

Comment:
You're right that AppModel can be gotten rid of, and the projectID parameter is misleading (I never actually handle `/redirect/project/projectID`) so I got rid of it too. (It was only there because I copied App::view to begin with, and then didn't quite realize that I could trim that parameter off).

The `if ($model->requireProject)` branch can also be simplified, because we're forcing the project to be `lexicon` and requireProject is always true for a lexicon project. And the `if ($model->isPublicApp)` part can go, because lexicon projects are not public apps. But the requireValidProjectIdForThisWebsiteAndValidateUserMembership call (and boy, is that a "moutfhul" of a function name) has to stay since that's where the lookup of the most recent project ID happens.

The new commit that I just pushed implements most of these suggestions, so that now the complexity should be manageable.